### PR TITLE
Step 9

### PR DESCRIPTION
**Pull request by devversion**
In progress

### CODE REVIEW

File: src/app/app.component.css [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+md-checkbox {
+  margin: 8px 0;
+}
+
+.form {
+  padding: 16px;
+}
+
+.form > * {
```
Comment (EladBezalel) [inline src/app/app.component.css:34]: Can't we use the fxLayoutGap here as well?

File: src/app/app.component.html [hunk 2]
```diff
@@ -1,3 +1,77 @@
...
+          <md-nav-list>
+            <md-list-item *ngFor="let user of users" (click)="selectedUser = user">
+              <md-icon svgIcon="avatars:{{user.avatar}}" class="avatar"></md-icon>
+              <span>{{user.name}}</span>
+            </md-list-item>
+          </md-nav-list>
+        </md-tab>
+        <md-tab label="Settings">
+
+          <form class="form" fxLayout="column" [formGroup]="userAddForm" (ngSubmit)="onSubmit()">
```
Comment (EladBezalel) [inline src/app/app.component.html:29]: do you mind explaining why `ngSubmit` and not the usual `submit`? 
> not saying it's wrong, just so i could explain to others

File: src/app/app.component.ts [hunk 3]
```diff
@@ -1,10 +1,104 @@
...
+  avatars = new Array(16).fill(0).map((_, i) => `svg-${i+1}`);
+
+  constructor(iconRegistry: MdIconRegistry, sanitizer: DomSanitizer, formBuilder: FormBuilder) {
+    // To avoid XSS attacks, the URL needs to be trusted from inside of your application.
+    const avatarsSafeUrl = sanitizer.bypassSecurityTrustResourceUrl('./assets/avatars.svg');
+
+    iconRegistry.addSvgIconSetInNamespace('avatars', avatarsSafeUrl);
+
+    this.userAddForm = formBuilder.group({
+      name: ['', Validators.required],
```
Comment (EladBezalel) [inline src/app/app.component.ts:87]: `required` attribute isn't enough?

File: src/app/app.module.ts [hunk 4]
```diff
@@ -1,18 +1,28 @@
 import {BrowserModule} from '@angular/platform-browser';
 import {NgModule} from '@angular/core';
-import {FormsModule} from '@angular/forms';
+import {FormsModule, ReactiveFormsModule} from '@angular/forms';
 import {HttpModule} from '@angular/http';
 
+import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
```
Comment (EladBezalel) [inline src/app/app.module.ts:6]: what for?

File: src/app/app.component.html (hunk 2 above)
Comment (devversion) [reply src/app/app.component.html:29]: Sure. `ngSubmit` only triggers if all Angular validators also pass.

File: src/app/app.component.ts (hunk 3 above)
Comment (devversion) [reply src/app/app.component.ts:87]: Not sure. IIRC for `ReactiveForms` you need to do it in the form group

File: src/app/app.module.ts (hunk 4 above)
Comment (devversion) [reply src/app/app.module.ts:6]: In Angular 4 you need either `BrowserAnimationsModule` or `BrowserNoopAnimationsModule`

**Pull request closed without merging**